stm32 clock mux for CK48 definition

FRASTM opened the pull request:
Define the STM32_CK48_ENABLED especially for the stm32F4 series

FIxes https://github.com/zephyrproject-rtos/zephyr/issues/94934

FRASTM committed:
include: drivers: stm32 clock mux CK48 definition

Define the STM32_CK48_ENABLED especially for the stm32F4 series
when ck48 node is enabled to leverage its already implemented support.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

etienne-lms commented on tests/drivers/clock_control/stm32_clock_configuration/stm32_common_devices/boards/f4_sdmmc48_pll.overlay:
```diff
@@ -12,6 +12,28 @@
+ * In case the clk48 is selecting the STM32_SRC_PLL_Q with
+ * clocks = <&rcc STM32_SRC_PLL_Q CK48M_SEL(0)>;
+ * one possible PLL config to give 48MHz on the pll_q output is as follows :
+ &pll {
+	div-m = <4>;
+	mul-n = <192>;
+	div-p = <4>;
+	div-q = <8>;
+	clocks = <&clk_hse>;
+	status = "okay";
+};
+
+&rcc {
+	clocks = <&pll>;
+	clock-frequency = <DT_FREQ_M(96)>;
+	ahb-prescaler = <1>;
+	apb1-prescaler = <2>;
+	apb2-prescaler = <1>;
+};
```
Intentional change?
Would it be worth to add an alternate test with its own DTS overlay file?

etienne-lms reviewed (dismissed):
LGTM.

Maybe the commit message could be more verbose on why the change, e.g.:
```patch
 Define the STM32_CK48_ENABLED especially for the stm32F4 series
+when ck48 node is enabled to leverage its already implemented support.
```

erwango reviewed (changes requested):
Nack until https://github.com/zephyrproject-rtos/zephyr/issues/94934#issuecomment-3390886046 is fixed

FRASTM committed:
drivers: clock control: stm32 common clock enables clk48

This commit enables the clk48 clock mux if STM32_CK48_ENABLED
is set by the device tree.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

etienne-lms reviewed (dismissed):
Comment below is not blocking.

etienne-lms reviewed (commented):
@erwango, could you update your review status? You https://github.com/zephyrproject-rtos/zephyr/pull/96353#pullrequestreview-3330344355 comment was addressed: see https://github.com/zephyrproject-rtos/zephyr/issues/94934#issuecomment-3400411928.
